static/js/disease/chart.ts

```diff
@@ -17,14 +17,253 @@ import * as d3 from "d3";
+      onMouseOver(idFunc(i), toolTipFunc(d), brightenPercentage);
+    })
+    .on("mousemove", onMouseMove)
+    .on("mouseout", (d, i) => onMouseOut(idFunc(i)));
+}
+
+/**
+ * Get function that takes an index and returns an ID containing the chart ID, element name, and index.
+ */
+function getElementIDFunc(
+  chartID: string,
+  elementName: string
+): (index: number) => string {
+  return (index) => `${chartID}-${elementName}${index}`;
+}
+/**
+ * Adds the x label to a graph based on user's input of width and height for label position, labelText for what the label reads, and svg for selecting the chart where the label is added
+ */
+function addXLabel(
+  width: number,
+  height: number,
+  labelText: string,
+  svg: d3.Selection<SVGGElement, unknown, HTMLElement, any>
+) {
+  svg
+    .attr("class", "axis-label")
+    .append("text")
+    .attr(
+      "transform",
+      "translate(" + width / 2 + " ," + (height + MARGIN.top + 40) + ")"
+    )
+    .text(labelText);
+}
+/**
+ * Adds the y label to a graph based on user's input of width and height for label position, labelText for what the label reads, and svg for selecting the chart where the label is added
+ */
+function addYLabel(
+  height: number,
+  labelText: string,
+  svg: d3.Selection<SVGGElement, unknown, HTMLElement, any>
+) {
+  svg
+    .append("text")
+    .attr("transform", "rotate(-90)")
+    .attr("y", 0 - MARGIN.left)
+    .attr("x", 0 - height / 2)
+    .attr("dy", "1em")
+    .text(labelText);
+}
+
+export interface DiseaseGeneAssociationRawData {
+  // name of the associated gene
   name: string;
+  // confidence score
   score: number;
+  // length of the confidence interval
+  interval: number;
 }
-
+export interface DiseaseGeneAssociationIntervalData {
+  // name of the associated gene
+  name: string;
+  // confidence score
+  score: number;
+  // lower confidence value = score - (interval/2)
+  lowerInterval: number;
+  // upper confidence value = score + (interval/2)
+  upperInterval: number;
+}
+/**
+ * Draws the disease-gene association charts for the disease of interest
+ * @param id
+ * @param data
+ * @returns
+ */
 export function drawDiseaseGeneAssocChart(
   id: string,
-  data: DiseaseGeneAssociationData[]
+  data: DiseaseGeneAssociationRawData[]
 ): void {
-  return;
+  // checks if the data is empty or not
+  if (_.isEmpty(data)) {
+    return;
+  }
+  let reformattedData = [] as DiseaseGeneAssociationIntervalData[];
+  // reformats the data by converting the confidence interval length to lower and upper confidence values
+  reformattedData = data.map((item) => {
+    return {
+      name: item.name,
+      score: Number(item.score),
+      lowerInterval: Number(item.score - item.interval / 2),
+      upperInterval: Number(item.score) + Number(item.interval / 2),
+    };
+  });
+  //Formats the gene name
+  function formatGeneName(d: string) {
+    // remove the word "bio/hg38_" from the entire gene name, say "bio/hg38_pRNA"
+    d = d.substring(9);
```

Comment:
nit: `d.replace("bio/hg38_", "")` is probably more readable.  I guess this is equivalent to `d.substring(9)` for strings that start with `"bio/hg38_"` because it only replaces the first instance.  But maybe you'll want to support other genome assemblies in the future, so perhaps `d.replace("bio\/hg\d{2}_", "")` or for the far future, `d.replace("bio\/hg\d+_", "")`.

Reply:
That's a great suggestion which is much more specific. Thank you, implemented it!